File: python/graphs/1557-minimum-number-of-vertices-to-reach-all-nodes.py
from typing import List


def findSmallestSetOfVertices(n: int, edges: List[List[int]]) -> List[int]:
    reachebles = set()
    for edge in edges:
        _, node = edge
        reachebles.add(node)
        
    response = []
    for i in range(n):
        if i not in reachebles:
            response.append(i)
            
    return response

File: cpp/graphs/1557-minimum-number-of-vertices-to-reach-all-nodes.cpp
#include <vector>

using namespace std;


vector<int> findSmallestSetOfVertices(int n, vector<vector<int>>& edges) {
    vector<int> not_reacheable;
    vector<int> indegree (n, 0);
    
    for (const auto edge : edges) {
        indegree[edge[1]]++;
    }
    
    for (int i = 0; i < n; i++) {
        if (indegree[i] == 0) {
            not_reacheable.push_back(i);
        }
    }
    
    return not_reacheable;
}